marlowe/src/Language/Marlowe/Core/V1/Semantics.hs

```diff
@@ -505,6 +520,10 @@ evalObservation env state obs = let
 refundOne :: Accounts -> Maybe ((Party, Token, Integer), Accounts)
 refundOne accounts = case Map.toList accounts of
     [] -> Nothing
+    -- SCP-5126: The return value of this function differs from
+    -- Isabelle semantics in that it returns the least-recently
+    -- added account-token combination rather than the first
+    -- lexicographically ordered one.
     ((accId, token), balance) : rest ->
         if balance > 0
         then Just ((accId, token, balance), Map.fromList rest)
```

Comment:
Should we say something about `Map.fromList . tail . Map.toList` preserving the invariant?

Reply:
Thanks! Fixed in e2277677987f359dcd8754275eccd5b7c9a880a7.